src/dsnotificationinterface.cpp

```diff
@@ -37,7 +44,9 @@ void CDSNotificationInterface::InitializeCurrentBlockTip()
 
 void CDSNotificationInterface::AcceptedBlockHeader(const CBlockIndex *pindexNew)
 {
-    llmq_ctx->clhandler->AcceptedBlockHeader(pindexNew);
+    assert(m_llmq_ctx);
```

Comment:
should `dsnotificationinferface` do assert? or just `return` enough? why assert?

Reply:
If it is accessing (smart) pointers, IMO yes. We check the validity of a (smart) pointer in other parts of the codebase before trying to access them and so far, we've implicitly assumed that in the time between the interface initializing and the first call being made, the dependent contexts/managers have been initialized.

Adding an assertion would alert us to anything that contravenes that assumption by explicitly defining it. For something like `BlockConnected`, the first call is to a manager in `m_llmq_ctx` so if a `BlockConnected` call is made that contravenes the assumption, we'd know about it immediately.

The same doesn't hold true for `UpdatedBlockTip`, which has some fast-fail conditions until we reach a possibly affected call. This no longer becomes a problem if we explicitly define assertions.

Plus, the assertion serve as a one-liner that enumerates the (smart) pointers being relied on for the rest of the function.